content/client-lib-development-guide/features.textile

```diff
@@ -696,3 +701,355 @@ The following default values are configured for the client library:
+  detach: () -> io ErrorInfo? // RTL5e
+  history: (
+    start: Time, // RTL10a
+    end: Time api-default now(), // RTL10a
+    direction: .Backwards | .Forwards api-default .Backwards, // RTL10a
+    limit: int api-default 100, // RTL10a
+    untilAttach: Bool default false // RTL10b
+  ) -> io PaginatedResult<Message> // RSL2a
+  subscribe: ((Message) ->) -> io // RTL7a
+  subscribe: (String, (Message) ->) -> io // RTL7b
+  unsubscribe: () // RTL8a, RTE5
+  unsubscribe: ((Message) ->) // RTL8a
+  unsubscribe: (String, (Message) ->) // RTL8a
+
+enum ChannelState:
+  INITIALIZED
+  ATTACHING
+  ATTACHED
+  DETACHING
+  DETACHED
+  FAILED
+
+class ChannelOptions:
+  encrypted: Bool // RSL5a, TB2a
+  cipherParams: CipherParams? // RSL5a, TB2b
+
+class CipherParams:
+  algorithm: String default "AES" // TZ2a
+  key: String // TZ2d
+  keyLength: Int? // TZ2b
+  mode: String default "CBC" // TZ2c
+
+class Crypto:
+  +getDefaultParams: (key: String?) -> CipherParams // TZ3
+
+class RestPresence:
+  get: (
+    limit: int api-default 100, // RSP3a
+    clientId: String?, // RSP3a2
+    connectionId: String?, // RSP3a3
+  ) -> io PaginatedResult<PresenceMessage> // RSPa
+  history: (
+    start: Time, // RSP4b1
+    end: Time api-default now(), // RSP4b1
+    direction: .Backwards | .Forwards api-default .Backwards, // RSP4b2
+    limit: int api-default 100, // RSP4b3
+  ) -> io PaginatedResult<PresenceMessage> // RSP4a
+
+class RealtimePresence:
+  syncComplete: Bool // RTP13
+  get: (
+    waitForSync: Bool default true, // RTP11c1
+    clientId: String?, // RTP11c2
+    connectionId: String?, // RTP11c3
+  ) -> io PaginatedResult<PresenceMessage> // RTP11
+  history: (
+    start: Time, // RTP12a
+    end: Time, // RTP12a
+    direction: .Backwards | .Forwards api-default .Backwards, // RTP12a
+    limit: int api-default 100, // RTP12a
+    untilAttach: Bool default false // RTP12b
+  ) -> io PaginatedResult<PresenceMessage> // RTP12c
+  subscribe: ((PresenceMessage) ->) -> io // RTP6a
+  subscribe: (PresenceAction, (PresenceMessage) ->) -> io // RTP6b
+  unsubscribe: () // RTP7a, RTE5
+  unsubscribe: ((PresenceMessage) ->) -> io // RTP7a
+  unsubscribe: (PresenceAction, (PresenceMessage) ->) -> io // RTP7b
+  // presence state modifiers
+  enter: (Data?) -> io ErrorInfo? // RTP8
+  update: (Data?) -> io ErrorInfo? // RTP9
+  leave: (Data?) -> io ErrorInfo? // RTP10
+  enterClient: (String, Data?) -> io ErrorInfo? // RTP4, RTP14, RTP15
+  updateClient: (String, Data?) -> io ErrorInfo? // RTP15
+  leaveClient: (String, Data?) -> io ErrorInfo? // RTP15
+
+enum PresenceAction:
+  ABSENT // TP2
+  PRESENT // TP2
+  ENTER // TP2
+  LEAVE // TP2
+  UPDATE // TP2
+
+class ConnectionDetails:
+  clientId: String? // RSA12a, CD2a
+  connectionKey: String // RTN15e, CD2b
+  connectionStateTtl: Duration // CD2f, RTN14e, DF1a
+  maxFrameSize: Int // CD2d
+  maxInboundRate: Int // CD2e
+  maxMessageSize: Int // CD2c
+  serverId: String // CD2g
+
+class Message:
```

Comment:
By the way, as this class can be instantiated by the user (as `Channel.publish` accepts `[Message]`) the constructors maybe should be also specified.